receiver/elasticsearchreceiver/integration_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package elasticsearchreceiver
+
+import (
+	"context"
+	"net"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/scrapertest"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/scrapertest/golden"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+)
+
+var (
+	containerRequest7_9_3 = testcontainers.ContainerRequest{
+		FromDockerfile: testcontainers.FromDockerfile{
+			Context:    filepath.Join("testdata", "integration"),
+			Dockerfile: "Dockerfile.elasticsearch.7_9_3",
+		},
+		ExposedPorts: []string{"9200:9200"},
+		WaitingFor: wait.ForListeningPort("9200").
+			WithStartupTimeout(2 * time.Minute),
+	}
+)
+
+func TestElasticsearchIntegration(t *testing.T) {
+	//Starts an elasticsearch docker container
+	t.Run("Running elasticsearch 7.9", func(t *testing.T) {
+		t.Parallel()
+		container := getContainer(t, containerRequest7_9_3)
+		defer func() {
+			require.NoError(t, container.Terminate(context.Background()))
+		}()
+		//todo
+		hostname, err := container.Host(context.Background())
+		require.NoError(t, err)
+
+		f := NewFactory()
+		cfg := f.CreateDefaultConfig().(*Config)
+		cfg.Endpoint = net.JoinHostPort(hostname, "9200")
```

Comment:
I figure out what is going on, this line set `cfg.Endpoint="localhost:9200"`, actullay, we need `http://localhost:9200`.